### DISCUSSION TITLE
Override the Default API Routes folder (/pages/api) so that actual pages can be hosted at that route

### DISCUSSION THREAD

**Discussion by spencer-stytch** [vercel/next.js#40157] [Ideas]
### Describe the feature you'd like to request

Currently the /pages/api route is reserved for Next API Routes. If I want to host pages on my app at domain/api/page-path, I need a workaround.

Rewrites is a possible workaround, but I am using next export, so it won't work.

exportPathMap is another potential workaround, but I have hundreds of pages hosted at domain/api/page-path because I have a file /pages/api/[[...slug]].tsx that leverages getStaticPaths and creates hundreds of pages.

I'd like a way to override the Next API Routes default path so that my /pages/api/[[...slug]].tsx file gets treated as any other file at a non-reserved route.

### Describe the solution you'd like

A config value in next.config.js that allows me to either turn off API Routes or change the default folder for them so that pages/api/page-file gets treated as other page files.

### Describe alternatives you've considered

rewrites, but that doesn't work because I'm using next export.

exportPathMap, but that's unsustainable because I have hundreds of pages in the app.

**Comment by softmarshmallow** [comment]
An updates?
